doc: Write the How do we Test and Document our C API? article

Hywan opened the pull request:
To view the article:

```sh
$ ./node_modules/.bin/next dev -p 3001
```

Then open http://localhost:3001/posts/how-do-we-test-and-document-our-c-api.

Alternatively, you can view it at https://github.com/wasmerio/wasmer.io/blob/d700d25ccb4a7b78f0644037540d11440c9d4f0c/_posts/how-do-we-test-and-document-our-c-api.md.

syrusakbary commented on _posts/how-do-we-test-and-document-our-c-api.md:
```diff
@@ -0,0 +1,420 @@
+published: true
+---
+
+[The Wasmer runtime](https://github.com/wasmerio/wasmer) comes in
+multiple flavors. Written in Rust, it is embedded in different
```
This will improve SEO for people searching for WebAssembly runtimes in Google.

```suggestion
Wasmer [server-side WebAssembly runtime](https://github.com/wasmerio/wasmer) comes in
multiple flavors. Written in Rust, it is embedded in different
```

syrusakbary commented on _posts/how-do-we-test-and-document-our-c-api.md:
```diff
@@ -0,0 +1,420 @@
+
+To run that test, one simply runs `cargo test --lib`.
+
+If you have ever written tests in C, you know the struggle. `CMake`,
+`Makefile`, you name it. What if we could write Rust code to test our
+C API? Something like:
```
```suggestion
If you have ever written tests in C you know the struggle. Either with `CMake`,
or `Makefile`, you name it.

But what if we could write Rust code to test our C API?
Something like:
```

syrusakbary replied:
```suggestion
If you have ever written tests in C you know the struggle. Either with `CMake`,
or `Makefile`, you name it.

But what if we could write Rust code to test our C API?
Something like:
```

Hywan commented:
Ready to be merged?

syrusakbary commented on _posts/how-do-we-test-and-document-our-c-api.md:
```diff
@@ -0,0 +1,422 @@
+---
+title: How do we Test and Document our C API?
```
This title gives very little context to someone that sees it for the first time (who is we, what is our stack in?)

Something more relatable should be:

```suggestion
title: Developing and Testing C libraries from Rust
```

syrusakbary reviewed (approved):
Once the title change is done, we should be good to merge and publish

Hywan commented:
I'm proposing this title:

> Novel way to Develop, Test and Document C libraries from Rust

I've updated the code accordingly. I've also fixed a typo.
